### PR TITLE
#30: Nozzle Server should optionally accept an external ActorSystem (closes #30)

### LINKED ISSUE #30: Nozzle Server should optionally accept an external ActorSystem
## stories
- As a service I would like to use only one ActorSystem to handle both http Server and http Client so that I can gracefully kill the Client ActorSystem if the Server fails (e.g. bind to port).
## requirements

Nozzle Server should optionally accept an external ActorSystem.

### PR DESCRIPTION
**Pull request by calippo**
Issue #30
### tests performed

tested on the example

### CODE REVIEW

File: src/main/scala/server/Server.scala [hunk 1]
```diff
@@ -152,19 +155,24 @@ class Server(
...
+      actorSystem,
       bootLog.logger,
       plainOldLoggerFactory.factory,
       config,
       router,
-      akkaAdditionalConf,
       bootTimeout)
 }
+
+sealed trait NozzleActorSystem
```
Comment (gabro) [inline src/main/scala/server/Server.scala:173]: why a custom class? Can't you just pass an `ActorSystem`?

File: src/main/scala/server/Server.scala [hunk 2]
```diff
@@ -54,28 +54,31 @@ class NozzleAkkaLogger extends akka.actor.Actor {
...
+  val systemName: String = actorSystem match {
+    case DefaultActorSystem(systemName, _) => systemName
+    case CustomActorSystem(as) => as.name
   }
 
   internal.akkaLoggers += systemName -> akkaLog
 
-  implicit val system: akka.actor.ActorSystem =
-    akka.actor.ActorSystem(systemName, internal.actorSystemLoggingConf.withFallback(akkaAdditionalConf))
+  implicit val system: akka.actor.ActorSystem = actorSystem match {
```
Comment (gabro) [inline src/main/scala/server/Server.scala:71]: not sold about this, why are we wrapping `ActorSystem` in another class?

File: src/main/scala/server/Server.scala (hunk 1 above)
Comment (calippo) [reply src/main/scala/server/Server.scala:173]: @gabro no

for backward compatibility, can't access `internal` from outside. For `DefaultActorSystem` I need the actor system to be defined inside nozzle. I think we need to review this whole module... but I don't want to break up the whole logging system for a small change.

```
akka.actor.ActorSystem(systemName, internal.actorSystemLoggingConf.withFallback(akkaAdditionalConf))
```

File: src/main/scala/server/Server.scala (hunk 2 above)
Comment (calippo) [reply src/main/scala/server/Server.scala:71]: see below, it's for backward compatibility

File: src/main/scala/server/Server.scala (hunk 1 above)
Comment (gabro) [reply src/main/scala/server/Server.scala:173]: I see. Ok then

**Pull request merged**